src/main/java/elon/Parser.java

```diff
@@ -0,0 +1,120 @@
+package elon;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+
+/**
+ * Parses user input commands and interacts with the UI, TaskList, and Storage.
+ */
+public class Parser {
+    private Ui ui;
+    private TaskList list;
+    private Scanner scanner;
+    private String[] inputArr;
+    private Storage storage;
+
+    /**
+     * Constructs a Parser object with the specified UI, TaskList, Scanner, input array, and Storage.
+     *
+     * @param ui         the UI object for interacting with the user
+     * @param list       the TaskList containing tasks
+     * @param scanner    the Scanner object for reading user input
+     * @param inputArr   the array of user input split into commands
+     * @param storage    the Storage object for saving and loading tasks
+     */
+    public Parser(Ui ui, TaskList list, Scanner scanner, String[] inputArr, Storage storage) {
+        this.ui = ui;
+        this.list = list;
+        this.scanner = scanner;
+        this.inputArr = inputArr;
+        this.storage = storage;
+    }
+
+    /**
+     * Reads the next line of user input and splits it into an array of strings.
+     *
+     * @param scanner the Scanner object used to read the input
+     * @return an array of strings representing the user input
+     */
+    public String[] nextInput(Scanner scanner) {
+        return scanner.nextLine().split(" ");
+    }
+    public void Parse() throws ElonException{
+        while (!inputArr[0].equals("bye")) {
+            if (inputArr[0].equals("list")) {
```

Comment:
Have you considered using an enumeration to match the inputs. Enums also play nice into using switch statements in my comment below. By using enumerations, it will improve readability as at one glance you be able to see all commands currently and allows for easier maintenance. But nice work overall!